lightning/src/chain/keysinterface.rs

```diff
@@ -181,6 +185,7 @@ impl_writeable_tlv_based_enum!(SpendableOutputDescriptor,
 	(0, StaticOutput) => {
 		(0, outpoint, required),
 		(2, output, required),
+		(4, channel_keys_id, option),
```

Comment:
Hmmmmmmmmmmmmmmmmmm, we really need to think harder about our forward-compat policy here - in general LDK strives to ensure you can always downgrade by a few releases and still have things work fine, but here we'd generate stuff that old versions don't know how to read/handle. I think we need to add a config option in the `KeysManager` that indicates we're doing "new style derivation", then only use the new derivation if its set. Also note that this will need to be odd, which means old versions will simply ignore it if they do not understand it (we'll need to write the flag to disk in the `InMemoryChannelKeys`'s so that old versions will refuse to read the data with the new flag set).

Reply:
hmm, how do we add a config option? Any examples you can point to? I'm assuming that by config you mean some file that is updated with some boolean value so that when you start the application that value is set for the `KeysManger`.

Reply:
Sorry for the response delay here, yea, I think we should just add a new field in `KeysManager`, with a parameter in the constructor to indicate that we don't need backwards compatibility with versions prior to whenever this gets merged.